apps/web/src/pages/releases/index.tsx

```diff
@@ -23,7 +26,28 @@ export default function Releases({ packs }: PublishedPacks) {
   const { t } = useTranslation()
   const language = useLanguage()
   const currency = useCurrency()
-  const { dispatch, state } = usePackFilter()
+  const { pathname, push, query } = useRouter()
+
+  // Get URL search params from router, stringify them...
+  const searchParams = stringify(query)
+  // ...so they can be processed by query-string
+  const initialState = parse(searchParams, {
+    parseBooleans: true,
+    parseNumbers: true,
+  })
+
+  // selectedOption is an object, so key off its id so it can live in URL
+  const initialSelectOptions = getSelectSortingOptions(t)
+  const selectedOption = getSelectSortingOptions(t).find(
+    (option) => option.id === initialState.selectedOption
+  )
+
+  // Set initial filter state based off of URL parsing
+  const { dispatch, state } = usePackFilter({
+    ...initialState,
+    selectedOption: selectedOption ? selectedOption : initialSelectOptions[0],
+  })
```

Comment:
Should some/most of this rely on `useMemo` to avoid recalculations/parsing each render?

Reply:
We could memoize the `find` but it's searching through 2 options so it feels like overkill.

Reply:
More thinking about the initial stringify/parse step. Also might help with avoiding the useEffect further down.

Reply:
I added memoization. The `useEffect` was necessary since it has to update once the state was changed... Open to other ideas though...?